packages/discovery-provider/src/challenges/tastemaker_challenge.py

```diff
@@ -0,0 +1,26 @@
+import logging
+from typing import Dict
+
+from sqlalchemy.orm.session import Session
+
+from src.challenges.challenge import ChallengeManager, ChallengeUpdater
+
+logger = logging.getLogger(__name__)
+
+
+class TastemakerChallengeUpdater(ChallengeUpdater):
+    """Tastemaker challenge
+
+    This challenge is completed when a user reposts or saves a track that later lands in the top 5 trending tracks.
+    """
+
+    def generate_specifier(self, session: Session, user_id: int, extra: Dict) -> str:
+        """
+        Generate a specifier for the challenge based on user_id and tastemaker_item_id
+        Format: {user_id}:{tastemaker_item_type}:{tastemaker_item_id}
+        """
+        item_type = "p" if extra["tastemaker_item_type"] == "playlist" else "t"
```

Comment:
just double checking here. if a user is a tastemaker for underground and the song later gets on overall trending, we only want them to get one reward right?

Reply:
good call. it appears that all tastemaker notifs right now only include user_id + track_id, so these will get dedup'd by the [`dedupe_notifications_from_existing_notifications` fn in `index_tastemaker_notifications.py`. ](https://github.com/AudiusProject/audius-protocol/blob/2a0405ee3a53235a03b8c9822462d656c98d1b00/packages/discovery-provider/src/tasks/index_tastemaker_notifications.py#L72). so i think we're good here.